my-app/src/App.js

```diff
@@ -0,0 +1,35 @@
+import React, { useState } from "react";
+import logo from "./logo.svg";
+import "./App.css";
+
+import Recipe from "./components/Recipe";
+import PrivateRoute from "./components/PrivateRoute";
+import {
+  BrowserRouter as Router,
+  Route,
+  useHistory,
+  Switch,
+} from "react-router-dom";
+import { Login } from "./components/Login";
+import Register from "./components/Register";
+
+function App() {
+  const [userId, setUserId] = useState("");
+  let history = useHistory();
+
+  return (
+  <div>
+        <Router>
+        <Route exact path="/">
+            <Login setUserId={setUserId} history={history} />
+          </Route>
+        <Route path="/register" component={Register} />
+        <PrivateRoute exact path="/:id/recipe" component ={Recipe} userId={userId}/>
```

Comment:
You can delete `userId={userId}`, the state isn't updating before this route gets called so it's not working, we are getting the number a different way.